exo-go/src/cmd/run.go

```diff
@@ -34,32 +35,43 @@ var runCmd = &cobra.Command{
 		if err != nil {
 			panic(err)
 		}
-		fmt.Printf("Running %s %s\n\n", appConfig.Name, appConfig.Version)
 		serviceNames := appConfigHelpers.GetServiceNames(appConfig.Services)
 		dependencyNames := appConfigHelpers.GetDependencyNames(appConfig)
 		silencedServiceNames := appConfigHelpers.GetSilencedServiceNames(appConfig.Services)
 		silencedDependencyNames := appConfigHelpers.GetSilencedDependencyNames(appConfig)
-
 		roles := append(serviceNames, dependencyNames...)
 		roles = append(roles, "exo-run")
 		logger := logger.NewLogger(roles, append(silencedServiceNames, silencedDependencyNames...), os.Stdout)
 
-		appRunner := appRunner.NewAppRunner(appConfig, logger, appDir, homeDir)
+		fmt.Printf("Setting up %s %s\n\n", appConfig.Name, appConfig.Version)
+		setup, err := appSetup.NewAppSetup(appConfig, logger, appDir, homeDir)
+		if err != nil {
+			panic(err)
+		}
+		err = setup.Setup()
+		if err != nil {
+			fmt.Println("setup failed")
+			panic(err)
```

Comment:
Instead of printing "setup failed" and then panic-ing with the error lets wrap the error and pass the wrapped error to panic